### ISSUE TITLE
Subclipse hangs when configured to use KWallet

### ISSUE THREAD

**Issue by andyg1001** [subclipse/subclipse#9]
I found a previous report [here](http://subversion.tigris.org/issues/show_bug.cgi?id=3498), in the old defunct issue tracker.  It seems this issue never got fully resolved.

SVN from the command line doesn't have any performance issues, but using Subclipse inside Eclipse will always cause the IDE to hang for about 30 to 60 seconds on first use in a session.  Worse, every now and then, on commit or update, Eclipse goes into a nose-dive, consumes in the order of 16 Gb of memory and causes the whole PC to grind to a halt requiring an ssh login from another computer to `kill -9` the process!

I am using Eclipse neon.1 and subclipse 1.10.13 which the newest I can install (the current "latest" repository will not install due to conflicting dependencies).  If I can give more helpful information, please direct me!

**Comment by markphip** [comment]
If it is related to KWallet, then it is unlikely we can do anything about it.  Most of the "real work" happens inside the Subversion API and we just have small Java high-level layer provided to us to call this API.  Here are some things I can suggest trying:

1. In Preferences under Team > SVN change to using SVNKit.  This is not my normal suggestion but it would be a good way to troubleshoot, since this is an alternate implementation of the API in pure Java, and does not even have any KWallet integration in the code (AFAIK).

2. If that helps, or even if it does not, it would be a good idea to switch over to our new version.  From the stand point of our code, it is largely the same but if we ever fix anything you will need to be on it anyway.  The main reason though, is that this version requires SVN 1.9.x which has been out for over a year.  Maybe whatever problem you are having has been fixed already?  This will require you to get the SVN 1.9.x client and libraries installed on your distro.  See:

https://github.com/subclipse/subclipse/wiki/JavaHL-on-Linux

In general I recommend using JavaHL when possible, since that is the native version provided by Subversion itself, but in your case, simply using SVNKit is likely to give you different results.  Worst case is it might give more insight into what the real problem is and then maybe it will be something we would know what to do to improve.

**Comment by andyg1001** [comment]
Unfortunately, swapping to SVNKit really doesn't work at all.  It seems to lose connection to the svn repository - by which I mean, you can no longer commit from the top-level directory (the menu item is even gone) and viewing changes from the local file to the repository happens instead against a blank file.  Returning to using JavaHL brings it all back on-line, but with the aforementioned problems.

For the record, my JavaHL is version 1.9.4 (r1740329) and SVNKit is version 1.9.0.10623.

It is possible that this is linked to the fact that I am using https:// to access the repository in question?  I don't seem to have the same problems with repositories using the svn:// protocol.

If JavaHL is simply a wrapper over the native subversion libraries, why should eclipse have problems that the command line tools do not?  What is really perplexing is the sudden desire of eclipse to consume gigabytes of memory sporadically during an update or commit.  That is the real killer for me; the 30 seconds "thinking time" at the beginning of the day is not an issue in comparison.  I also need to "think" (i.e. get a coffee) at the beginning of the day too.

**Comment by markphip** [comment]
There is no reason those SVNKit things you mention should happen.  I am not aware of any significant differences between the API.  That said, I am not aware of SVNKit 1.9.0 even existing yet.  The latest version on their website is 1.8.14 and I think 1.8.13 is the one we had on our website.  Maybe some new bug in their library?

FWIW, I use https:// exclusively.

As for JavaHL, the libraries are used a lot different than command line since the command line process always terminates when command does.  Also, Subversion provides a C++ layer which is the libsvnjavahl-1 native part that the Java JNI shim talks to and this also maintains some state etc.

You can also try disabling KWallet.  If you want to use it for the command line client, but just disabled it for Subclipse you can make a copy of your ~/.subversion folder and then point Subclipse to the copy in the preferences.  Then edit the config file in the copy to disable the password stores:

```
[auth]
password-stores =

```

If the problem is with the SVN KWallet support, then this would eliminate that.

**Comment by andyg1001** [comment]
Yes, you're right -- there isn't a SVNKit 1.9.0.  It seems that it is really 1.8.13.10623 according to the list of installed packages, but under Preferences > Team > SVN > SVN interface, it reports itself as 1.9.0.10623.  Strange!

Disabling KWallet support does indeed eliminate the problem inside eclipse, but this isn't really a long-term option as I am not permitted by company policy to store these passwords unencrypted.  The alternative is to enter the password every time, but it is not a particularly memorable one, being a string of completely random characters!  I would rather remove svn support from eclipse and just use the command line, but that too would be a great shame as I rely on the functionality inside eclipse a lot.

Is it possible to diagnose, somehow, why and where the extreme memory consumption takes place?

**Comment by markphip** [comment]
I do not know how to diagnose it further.  You would have to move to the Subversion users@ list making it clear this is about JavaHL via Subclipse and share some of the info you can gather about the problem.  Then hopefully someone will have some suggestions of what to do next to gather even more info that might help.

I would still consider SVNKit too.  It should not have these specific problems, and there is no reason it should not be working correctly either.  It might be worth doing a clean checkout using SVNKit in your preferences and see if you run into trouble.  The 1.9 working copy format is supposed to be compatible but maybe there is some subtle difference about what native SVN is doing that is breaking SVNKit.  It also has its own pure Java SQLite (which is where SVN stores the metadata).  Maybe you have a more recent native SQLite that has subtle differences too.  So a fully clean and pure SVNKit working copy *might* not show same issues.
